src/components/TodoModal/TodoModal.tsx

```diff
@@ -15,25 +33,34 @@
               className="modal-card-title has-text-weight-medium"
               data-cy="modal-header"
             >
-              Todo #2
+              {`Todo #${selectTodo.id}`}
             </div>
 
             {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-            <button type="button" className="delete" data-cy="modal-close" />
+            <button
+              type="button"
+              className="delete"
+              data-cy="modal-close"
+              onClick={() => setSelectTodo(null)}
+            />
```

Comment:
The close button lacks an associated label, which might affect accessibility. Consider adding an aria-label to the button to describe its action, such as 'Close modal'.